### ISSUE TITLE
-Q new always fails

### ISSUE THREAD

**Issue by hansiang93** [jython/jython#27]
When running jython with -Qnew, it always errors out.

Looks like there should be a break after new in the switch case to prevent the -Q option from falling through to the default failure

**Comment by hansiang93** [comment]
@jeff5 are we using github for merges of PRs now? I'd love to contribute if that's the case~

**Comment by jeff5** [comment]
Yes, and you'd be welcome. Fork this repo please, not the frozen mirror.

I was going to leave it a couple more days before inviting new issues and PRs here, in case someone spots that I've done the migration wrong. But if you want to be the first to test [the process](https://jython-devguide.readthedocs.io/en/latest/), go ahead.

For anything more than a trivial fix, we ask you to fill in the [PSF CLA](https://www.python.org/psf/contrib/contrib-form/). For this you need an account on the [CPython bug-tracker](https://bugs.python.org/), then fill in the form at the link.

One break statement is trivial, but maybe you mean to do more? A test that would have caught my silly mistake would be a good addition.

**Comment by hansiang93** [comment]
i've found a few other small issues while porting over to 2.7.2, and had made the patches locally to work around them while hoping to put up some PRs against the initial Jythontools git. But since the project had been moved over to git officially, i thought to wait till after before i get started on them :)

**Comment by jeff5** [comment]
I'm content that the related PR fixes this. What is bugging me (apart from the silly mistake) is that no test failed. This should have caught it:
https://github.com/jython/jython/blob/master/Lib/test/test_cmd_line.py#L44-L48

But I can see why it didn't here:
https://github.com/jython/jython/blob/master/Lib/test/test_cmd_line.py#L37
Anything that ends with a newline is acceptable, even an error message.

Now, we copied this from CPython, and that test appears to be broken there. In fact, I can do this in CPython 2.7.16:
```
>>> import test.test_cmd_line
>>> import test.test_support
>>> class MyTest(test.test_cmd_line.CmdLineTest):
...     def test_q2(self):
...         self.verify_valid_flag('-Qbad')
...
>>> test.test_support.run_unittest(MyTest)
...
test_q2 (__main__.MyTest) ... ok
...
----------------------------------------------------------------------
Ran 13 tests in 7.733s

OK
```
What I'm doing here is sub-classing the test case and adding a test that ought to fail, but doesn't when I run it as the test framework would.

I think this bug is still present in CPython 3. `-Q` is gone, and the code supporting the test is different:
https://github.com/python/cpython/blob/6a1e9c26736259413b060b01d1cb52fcf82c0385/Lib/test/test_cmd_line.py#L34-L38
but I find I can get it to accept any nonsense as an option.
